android/jni/tun2socks/tun2socks.c

```diff
@@ -1477,22 +1500,35 @@ int process_device_udp_packet (uint8_t *data, int data_len)
     BAddr_Print(&local_addr, local_addr_str);
     char remote_addr_str[BADDR_MAX_PRINT_LEN];
     BAddr_Print(&remote_addr, remote_addr_str);
-    BLog(BLOG_INFO, "UDP: %s -> %s. DNS: %d", local_addr_str, remote_addr_str, is_dns);
```

Comment:
Expanding the folded code above, I see the `switch (ip_version)`, for only two meaningful cases (4 and 6), both of which do similar things (parse header, parse udp, check integrity). I know this is less frowned upon in C, but is there an idiomatic way to refactor this into a less repetitive and procedural spaghetti style? (Btw, the original switch statement was found to have been [misdesigned](http://stackoverflow.com/a/252733). Thankfully they're redesigned in Swift and other modern languages. If we're going to use them in C, all the more reason to set up a linter to detect and warn on possibly-unintentional fall-through.)